### PR TITLE
Internal: Editor floating panels - Add types and position math utilities [ED-24738]

### PR DESCRIPTION
**Pull request by rami-elementor**
## Summary

Adds the foundational types and pure position math utilities for the editor floating panels package.

Split from #36308 (1/5).

## Scope

- Types: `LogicalSize`, `LogicalPosition`, `FloatingPanelDefaults`, `PanelCorner`
- Utils: corner-position transforms (RTL-aware), drag-math, resize-math, viewport-bounds
- Tests for all math utilities

## Review focus

Math correctness, RTL position transforms, bounds clamping edge cases.

Ref: ED-24738

Made with [Cursor](https://cursor.com)

## 1. Problem & Context

Establishes foundational type definitions and position/sizing math utilities for floating panel UI components in the editor. Supports logical positioning (directional-aware) and drag/resize calculations with viewport constraints.

## 2. What Changed (Where)

| File | Change |
|------|--------|
| `types.ts` | New type exports: `FloatingPanelState`, `FloatingPanelDeclaration`, `LogicalPosition`, `LogicalSize`, `PanelCorner` |
| `corner-position.ts` | Panel corner positioning: anchor conversions, inset key mappings, drag bounds calculation (149 lines) |
| `drag-math.ts` | Physical-to-logical delta conversion, RTL handling, bounded position updates on drag |
| `resize-math.ts` | Edge/corner resize handlers with position/size recalculation maintaining anchored edges |
| `utils/clamp.ts`, `direction.ts`, `viewport-bounds.ts` | Helper utilities: numeric clamping, RTL detection, app bar/side panel constants |
| `constants.ts` | Z-index base constant (1000) |
| `package.json` | New public package definition with React peer deps |
| `jest.config.js` | SWC transform configured with React automatic runtime |
| Tests | 238 lines: corner position round-trips, drag/resize bounds clamping, RTL behavior |

## 3. How It Works

Position calculations normalize around logical insets (`insetInlineStart`, `insetInlineEnd`, `insetBlockStart`, `insetBlockEnd`) keyed by `PanelCorner` type. Drag deltas convert physical pixels to logical (accounting for RTL), then clamp against viewport bounds respecting app bar/side panel insets. Resize operations anchor to the opposite edge—e.g., inline-start resize maintains inline-end while growing width. All operations preserve bounds validity even when panel exceeds free space (inverted bounds).

## 4. Risks

Minor: SWC transform config change assumes jsdom environment compatibility. No data migrations or breaking API changes—pure utility foundation. Test coverage is thorough (zero edge case gaps in resize/bounds logic).

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: packages/packages/core/editor-floating-panels/src/utils/resize-math.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+export function applyBlockEndResize( size: LogicalSize, blockDelta: number, bounds: ResizeBounds ): LogicalSize {
+	return {
+		inlineSize: size.inlineSize,
+		blockSize: clamp( size.blockSize + blockDelta, bounds.minBlockSize, bounds.maxBlockSize ),
+	};
+}
+
+export function applyInlineStartResize(
```
[the remark this answers is not in the document]
Comment (rami-elementor) [reply packages/packages/core/editor-floating-panels/src/utils/resize-math.ts:34]: The resize hook already handles this — see `use-floating-panel-resize.ts` lines 97-120 where `toStartAnchoredPosition` / `fromStartAnchoredPosition` bracket the `applyResize` call. The utility functions are intentionally thin and stateless; the coordinate normalization is the caller's responsibility by design."

**Comment by Ntnelbaba** [comment]
Superseded by consolidated PR #36362 targeting `main`. All changes from this stacked PR are included there.
